localstack/services/sns/provider.py

```diff
@@ -888,7 +887,8 @@ async def message_to_subscriber(
                 "Unable to run Lambda function on SNS message: %s %s", exc, traceback.format_exc()
             )
             store_delivery_log(subscriber, False, message, message_id)
-            sns_error_to_dead_letter_queue(subscriber["SubscriptionArn"], req_data, str(exc))
+            message_body = create_sns_message_body(subscriber, req_data, message_id)
+            sns_error_to_dead_letter_queue(subscriber["SubscriptionArn"], message_body, str(exc))
```

Comment:
I think this line is not necessary (since it's the same as in line 807 and the params haven't changed I guess), right?
```suggestion
            sns_error_to_dead_letter_queue(subscriber["SubscriptionArn"], message_body, str(exc))
```

Reply:
The 807 line is in a different `if` condition than line 890. But we don't need to create the message body if if we are in the `subscriber["Protocol"] == "sms"` block or in the `elif subscriber["Protocol"] in ["email", "email-json"]` when the protocol isn't "email-json". Maybe I can create the message body before all `if...elif` blocks, but it might not be used. What do you think?

Reply:
Ah, I see. I was a bit confused by GitHub collapsing parts of the code. No need to change anything then. :)